packages/blade/src/components/Table/TablePagination.web.tsx

```diff
@@ -212,7 +213,16 @@ const _TablePagination = ({
   }, [controlledCurrentPage, currentPage, handlePageChange, onPageChange]);
 
   if (currentPage > totalPages - 1) {
-    handlePageChange(totalPages - 1);
+    if (!isUndefined(controlledCurrentPage)) {
```

Comment:
```suggestion
    if (controlledCurrentPage) {
```

this is simpler to read I guess 🤔

Reply:
controlledCurrentPage can be 0 or 1 which is a valid boolean 😅

Reply:
ohhh